{
    "Modified": "2012-10-30T23:06:05.607000",
    "Published": "2008-10-31T20:00:01.057000",
    "access": {
        "authentication": "NONE",
        "complexity": "LOW",
        "vector": "NETWORK"
    },
    "capec": [],
    "cvss": "7.5",
    "cvss-time": "2008-11-03T13:29:00",
    "cwe": "CWE-189",
    "exploit-db": [
        {
            "description": "Python 2.5.2 'Imageop' Module Argument Validation Buffer Overflow Vulnerability. CVE-2008-4864. Dos exploit for unix platform",
            "id": "EDB-ID:32534",
            "last seen": "2016-02-03",
            "modified": "2008-10-27",
            "published": "2008-10-27",
            "reporter": "Chris Evans",
            "source": "https://www.exploit-db.com/download/32534/",
            "title": "Python <= 2.5.2 - 'Imageop' Module Argument Validation Buffer Overflow Vulnerability"
        },
        {
            "description": "Python < 2.5.2 Imageop Module - 'imageop.crop()' Buffer Overflow Vulnerability. CVE-2008-4864. Dos exploits for multiple platform",
            "id": "EDB-ID:10229",
            "last seen": "2016-02-01",
            "modified": "2009-11-24",
            "published": "2009-11-24",
            "reporter": "Chris Evans",
            "source": "https://www.exploit-db.com/download/10229/",
            "title": "Python < 2.5.2 Imageop Module - 'imageop.crop' Buffer Overflow Vulnerability"
        }
    ],
    "id": "CVE-2008-4864",
    "impact": {
        "availability": "PARTIAL",
        "confidentiality": "PARTIAL",
        "integrity": "PARTIAL"
    },
    "last-modified": "2018-10-11T16:53:01.717000",
    "nessus": [
        {
            "NASL family": "SuSE Local Security Checks",
            "NASL id": "SUSE_PYTHON-5837.NASL",
            "description": "Integer Overflows in the python imageop module and in the expandtabs method potentially allowed attackers to execute arbitrary code.\n(CVE-2008-4864 / CVE-2008-5031)",
            "last seen": "2019-02-21",
            "modified": "2012-05-17",
            "plugin id": "41581",
            "published": "2009-09-24",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=41581",
            "title": "SuSE 10 Security Update : Python (ZYPP Patch Number 5837)"
        },
        {
            "NASL family": "Ubuntu Local Security Checks",
            "NASL id": "UBUNTU_USN-806-1.NASL",
            "description": "It was discovered that Python incorrectly handled certain arguments in the imageop module. If an attacker were able to pass specially crafted arguments through the crop function, they could execute arbitrary code with user privileges. For Python 2.5, this issue only affected Ubuntu 8.04 LTS. (CVE-2008-4864)\n\nMultiple integer overflows were discovered in Python's stringobject and unicodeobject expandtabs method. If an attacker were able to exploit these flaws they could execute arbitrary code with user privileges or cause Python applications to crash, leading to a denial of service. (CVE-2008-5031).\n\nNote that Tenable Network Security has extracted the preceding description block directly from the Ubuntu security advisory. Tenable has attempted to automatically clean and format it as much as possible without introducing additional issues.",
            "last seen": "2019-02-21",
            "modified": "2018-11-28",
            "plugin id": "40361",
            "published": "2009-07-24",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=40361",
            "title": "Ubuntu 6.06 LTS / 8.04 LTS / 8.10 : python2.4, python2.5 vulnerabilities (USN-806-1)"
        },
        {
            "NASL family": "SuSE Local Security Checks",
            "NASL id": "SUSE_11_0_PYTHON-081201.NASL",
            "description": "Integer Overflows in the python imageop module potentially allowed attackers to execute arbitrary code (CVE-2008-4864).",
            "last seen": "2019-02-21",
            "modified": "2014-06-13",
            "plugin id": "40116",
            "published": "2009-07-21",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=40116",
            "title": "openSUSE Security Update : python (python-360)"
        },
        {
            "NASL family": "SuSE Local Security Checks",
            "NASL id": "SUSE9_12316.NASL",
            "description": "Integer overflows in the python imageop module and in the expandtabs method potentially allowed attackers to execute arbitrary code.\n(CVE-2008-4864, CVE-2008-5031)",
            "last seen": "2019-02-21",
            "modified": "2012-04-23",
            "plugin id": "41260",
            "published": "2009-09-24",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=41260",
            "title": "SuSE9 Security Update : Python (YOU Patch Number 12316)"
        },
        {
            "NASL family": "MacOS X Local Security Checks",
            "NASL id": "MACOSX_SECUPD2009-001.NASL",
            "description": "The remote host is running a version of Mac OS X 10.5 or 10.4 that does not have Security Update 2009-001 applied.\n\nThis security update contains fixes for the following products :\n\n  - AFP Server\n  - Apple Pixlet Video\n  - CarbonCore\n  - CFNetwork\n  - Certificate Assistant\n  - ClamAV\n  - CoreText\n  - CUPS\n  - DS Tools\n  - fetchmail\n  - Folder Manager\n  - FSEvents\n  - Network Time\n  - perl\n  - Printing\n  - python\n  - Remote Apple Events\n  - Safari RSS\n  - servermgrd\n  - SMB\n  - SquirrelMail\n  - X11\n  - XTerm",
            "last seen": "2019-02-21",
            "modified": "2018-07-16",
            "plugin id": "35684",
            "published": "2009-02-13",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=35684",
            "title": "Mac OS X Multiple Vulnerabilities (Security Update 2009-001)"
        },
        {
            "NASL family": "Misc.",
            "NASL id": "VMWARE_VMSA-2009-0016_REMOTE.NASL",
            "description": "The remote VMware ESX / ESXi host is missing a security-related patch.\nIt is, therefore, affected by multiple vulnerabilities, including remote code execution vulnerabilities, in the following components :\n\n  - Apache Geronimo\n  - Apache Tomcat\n  - Apache Xerces2\n  - cURL/libcURL\n  - ISC BIND\n  - Libxml2\n  - Linux kernel\n  - Linux kernel 64-bit\n  - Linux kernel Common Internet File System\n  - Linux kernel eCryptfs\n  - NTP\n  - Python\n  - Java Runtime Environment (JRE)\n  - Java SE Development Kit (JDK)\n  - Java SE Abstract Window Toolkit (AWT)\n  - Java SE Plugin\n  - Java SE Provider\n  - Java SE Swing\n  - Java SE Web Start",
            "last seen": "2019-02-21",
            "modified": "2018-08-06",
            "plugin id": "89117",
            "published": "2016-03-03",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=89117",
            "title": "VMware ESX / ESXi Multiple Vulnerabilities (VMSA-2009-0016) (remote check)"
        },
        {
            "NASL family": "SuSE Local Security Checks",
            "NASL id": "SUSE_PYTHON-5848.NASL",
            "description": "Integer Overflows in the python imageop module potentially allowed attackers to execute arbitrary code (CVE-2008-4864).",
            "last seen": "2019-02-21",
            "modified": "2014-06-13",
            "plugin id": "35332",
            "published": "2009-01-11",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=35332",
            "title": "openSUSE 10 Security Update : python (python-5848)"
        },
        {
            "NASL family": "VMware ESX Local Security Checks",
            "NASL id": "VMWARE_VMSA-2009-0016.NASL",
            "description": "a. JRE Security Update\n\n  JRE update to version 1.5.0_20, which addresses multiple security   issues that existed in earlier releases of JRE.\n\n  The Common Vulnerabilities and Exposures project (cve.mitre.org) has   assigned the following names to the security issues fixed in   JRE 1.5.0_18: CVE-2009-1093, CVE-2009-1094, CVE-2009-1095,   CVE-2009-1096, CVE-2009-1097, CVE-2009-1098, CVE-2009-1099,   CVE-2009-1100, CVE-2009-1101, CVE-2009-1102, CVE-2009-1103,   CVE-2009-1104, CVE-2009-1105, CVE-2009-1106, and CVE-2009-1107.\n\n  The Common Vulnerabilities and Exposures project (cve.mitre.org) has   assigned the following names to the security issues fixed in   JRE 1.5.0_20: CVE-2009-2625, CVE-2009-2670, CVE-2009-2671,   CVE-2009-2672, CVE-2009-2673, CVE-2009-2675, CVE-2009-2676,   CVE-2009-2716, CVE-2009-2718, CVE-2009-2719, CVE-2009-2720,   CVE-2009-2721, CVE-2009-2722, CVE-2009-2723, CVE-2009-2724.\n\nb. Update Apache Tomcat version\n\n  Update for VirtualCenter and ESX patch update the Tomcat package to   version 6.0.20 (vSphere 4.0) or version 5.5.28 (VirtualCenter 2.5)   which addresses multiple security issues that existed   in the previous version of Apache Tomcat.\n\n  The Common Vulnerabilities and Exposures project (cve.mitre.org) has   assigned the following names to the security issues fixed in   Apache Tomcat 6.0.20 and Tomcat 5.5.28: CVE-2008-5515,   CVE-2009-0033, CVE-2009-0580, CVE-2009-0781, CVE-2009-0783.\n\n  The Common Vulnerabilities and Exposures project (cve.mitre.org) has   assigned the following names to the security issues fixed in   Apache Tomcat 6.0.18:  CVE-2008-1232, CVE-2008-1947, CVE-2008-2370.\n\n  The Common Vulnerabilities and Exposures project (cve.mitre.org) has   assigned the following names to the security issues fixed in   Apache Tomcat 6.0.16:  CVE-2007-5333, CVE-2007-5342, CVE-2007-5461,   CVE-2007-6286, CVE-2008-0002.\n  c. Third-party library update for ntp.\n   The Network Time Protocol (NTP) is used to synchronize a computer's   time with a referenced time source.\n   ESXi 3.5 and ESXi 4.0 have a ntp client that is affected by the   following security issue. Note that the same security issue is   present in the ESX Service Console as described in section d. of   this advisory.\n   A buffer overflow flaw was discovered in the ntpd daemon's NTPv4   authentication code. If ntpd was configured to use public key   cryptography for NTP packet authentication, a remote attacker could   use this flaw to send a specially crafted request packet that could   crash ntpd or, potentially, execute arbitrary code with the   privileges of the 'ntp' user.\n   The Common Vulnerabilities and Exposures Project (cve.mitre.org)   has assigned the name CVE-2009-1252 to this issue.\n   The NTP security issue identified by CVE-2009-0159 is not relevant   for ESXi 3.5 and ESXi 4.0.\n d. Service Console update for ntp\n\n  Service Console package ntp updated to version ntp-4.2.2pl-9el5_3.2    The Network Time Protocol (NTP) is used to synchronize a computer's   time with a referenced time source.\n   The Service Console present in ESX is affected by the following   security issues.\n   A buffer overflow flaw was discovered in the ntpd daemon's NTPv4   authentication code. If ntpd was configured to use public key   cryptography for NTP packet authentication, a remote attacker could   use this flaw to send a specially crafted request packet that could   crash ntpd or, potentially, execute arbitrary code with the   privileges of the 'ntp' user.\n   NTP authentication is not enabled by default on the Service Console.\n   The Common Vulnerabilities and Exposures Project (cve.mitre.org)   has assigned the name CVE-2009-1252 to this issue.\n   A buffer overflow flaw was found in the ntpq diagnostic command. A   malicious, remote server could send a specially crafted reply to an   ntpq request that could crash ntpq or, potentially, execute   arbitrary code with the privileges of the user running the ntpq   command.\n   The Common Vulnerabilities and Exposures Project (cve.mitre.org)   has assigned the name CVE-2009-0159 to this issue.\n  e. Updated Service Console package kernel\n\n  Updated Service Console package kernel addresses the security   issues listed below.\n\n  The Common Vulnerabilities and Exposures project (cve.mitre.org)   has assigned the names CVE-2008-3528, CVE-2008-5700, CVE-2009-0028,   CVE-2009-0269, CVE-2009-0322, CVE-2009-0675, CVE-2009-0676,   CVE-2009-0778 to the security issues fixed in kernel   2.6.18-128.1.6.\n\n  The Common Vulnerabilities and Exposures project (cve.mitre.org)   has assigned the names CVE-2008-4307, CVE-2009-0834, CVE-2009-1337,   CVE-2009-0787, CVE-2009-1336 to the security issues fixed in   kernel 2.6.18-128.1.10.\n\n  The Common Vulnerabilities and Exposures project (cve.mitre.org)   has assigned the names CVE-2009-1439, CVE-2009-1633, CVE-2009-1072,   CVE-2009-1630, CVE-2009-1192 to the security issues fixed in   kernel 2.6.18-128.1.14.\n\n  The Common Vulnerabilities and Exposures project (cve.mitre.org)   has assigned the names CVE-2007-5966, CVE-2009-1385, CVE-2009-1388,   CVE-2009-1389, CVE-2009-1895, CVE-2009-2406, CVE-2009-2407 to the   security issues fixed in kernel 2.6.18-128.4.1.\n\n  The Common Vulnerabilities and Exposures project (cve.mitre.org)   has assigned the names CVE-2009-2692, CVE-2009-2698 to the   security issues fixed in kernel 2.6.18-128.7.1.\n\n  The Common Vulnerabilities and Exposures project (cve.mitre.org)   has assigned the names CVE-2009-0745, CVE-2009-0746, CVE-2009-0747,   CVE-2009-0748, CVE-2009-2847, CVE-2009-2848 to the security issues   fixed in kernel 2.6.18-164.\n\n f. Updated Service Console package python\n\n  Service Console package Python update to version 2.4.3-24.el5.\n\n  When the assert() system call was disabled, an input sanitization   flaw was revealed in the Python string object implementation that   led to a buffer overflow. The missing check for negative size values   meant the Python memory allocator could allocate less memory than   expected. This could result in arbitrary code execution with the   Python interpreter's privileges.\n\n  Multiple buffer and integer overflow flaws were found in the Python   Unicode string processing and in the Python Unicode and string   object implementations. An attacker could use these flaws to cause   a denial of service.\n\n  Multiple integer overflow flaws were found in the Python imageop   module. If a Python application used the imageop module to   process untrusted images, it could cause the application to   disclose sensitive information, crash or, potentially, execute   arbitrary code with the Python interpreter's privileges.\n\n  Multiple integer underflow and overflow flaws were found in the   Python snprintf() wrapper implementation. An attacker could use   these flaws to cause a denial of service (memory corruption).\n\n  Multiple integer overflow flaws were found in various Python   modules. An attacker could use these flaws to cause a denial of   service.\n\n  An integer signedness error, leading to a buffer overflow, was   found in the Python zlib extension module. If a Python application   requested the negative byte count be flushed for a decompression   stream, it could cause the application to crash or, potentially,   execute arbitrary code with the Python interpreter's privileges.\n\n  A flaw was discovered in the strxfrm() function of the Python   locale module. Strings generated by this function were not properly   NULL-terminated, which could possibly cause disclosure of data   stored in the memory of a Python application using this function.\n\n  The Common Vulnerabilities and Exposures project (cve.mitre.org)   has assigned the names CVE-2007-2052 CVE-2007-4965 CVE-2008-1721   CVE-2008-1887 CVE-2008-2315 CVE-2008-3142 CVE-2008-3143   CVE-2008-3144 CVE-2008-4864 CVE-2008-5031 to these issues.\n\n g. Updated Service Console package bind\n\n  Service Console package bind updated to version 9.3.6-4.P1.el5\n\n  The Berkeley Internet Name Domain (BIND) is an implementation of the   Domain Name System (DNS) protocols. BIND includes a DNS server   (named); a resolver library (routines for applications to use when   interfacing with DNS); and tools for verifying that the DNS server   is operating correctly.\n\n  A flaw was found in the way BIND handles dynamic update message   packets containing the 'ANY' record type. A remote attacker could   use this flaw to send a specially crafted dynamic update packet   that could cause named to exit with an assertion failure.\n\n  The Common Vulnerabilities and Exposures project (cve.mitre.org)   has assigned the name CVE-2009-0696 to this issue.\n\n h. Updated Service Console package libxml2\n\n  Service Console package libxml2 updated to version 2.6.26-2.1.2.8.\n\n  libxml is a library for parsing and manipulating XML files. A   Document Type Definition (DTD) defines the legal syntax (and also   which elements can be used) for certain types of files, such as XML   files.\n\n  A stack overflow flaw was found in the way libxml processes the   root XML document element definition in a DTD. A remote attacker   could provide a specially crafted XML file, which once opened by a   local, unsuspecting user, would lead to denial of service.\n\n  Multiple use-after-free flaws were found in the way libxml parses   the Notation and Enumeration attribute types. A remote attacker   could provide a specially crafted XML file, which once opened by a   local, unsuspecting user, would lead to denial of service.\n\n  The Common Vulnerabilities and Exposures project (cve.mitre.org)   has assigned the names CVE-2009-2414 and CVE-2009-2416 to these   issues.\n\n i. Updated Service Console package curl\n\n  Service Console package curl updated to version 7.15.5-2.1.el5_3.5\n\n  A cURL is affected by the previously published 'null prefix attack',   caused by incorrect handling of NULL characters in X.509   certificates. If an attacker is able to get a carefully-crafted   certificate signed by a trusted Certificate Authority, the attacker   could use the certificate during a man-in-the-middle attack and   potentially confuse cURL into accepting it by mistake.\n\n  The Common Vulnerabilities and Exposures project (cve.mitre.org)   has assigned the name CVE-2009-2417 to this issue\n\n j. Updated Service Console package gnutls\n\n  Service Console package gnutil updated to version 1.4.1-3.el5_3.5\n\n  A flaw was discovered in the way GnuTLS handles NULL characters in   certain fields of X.509 certificates. If an attacker is able to get   a carefully-crafted certificate signed by a Certificate Authority   trusted by an application using GnuTLS, the attacker could use the   certificate during a man-in-the-middle attack and potentially   confuse the application into accepting it by mistake.\n\n  The Common Vulnerabilities and Exposures project (cve.mitre.org)   has assigned the name CVE-2009-2730 to this issue",
            "last seen": "2019-02-21",
            "modified": "2018-08-06",
            "plugin id": "42870",
            "published": "2009-11-23",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=42870",
            "title": "VMSA-2009-0016 : VMware vCenter and ESX update release and vMA patch release address multiple security issues in third party components."
        },
        {
            "NASL family": "CentOS Local Security Checks",
            "NASL id": "CENTOS_RHSA-2009-1176.NASL",
            "description": "Updated python packages that fix multiple security issues are now available for Red Hat Enterprise Linux 5.\n\nThis update has been rated as having moderate security impact by the Red Hat Security Response Team.\n\nPython is an interpreted, interactive, object-oriented programming language.\n\nWhen the assert() system call was disabled, an input sanitization flaw was revealed in the Python string object implementation that led to a buffer overflow. The missing check for negative size values meant the Python memory allocator could allocate less memory than expected. This could result in arbitrary code execution with the Python interpreter's privileges. (CVE-2008-1887)\n\nMultiple buffer and integer overflow flaws were found in the Python Unicode string processing and in the Python Unicode and string object implementations. An attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-3142, CVE-2008-5031)\n\nMultiple integer overflow flaws were found in the Python imageop module. If a Python application used the imageop module to process untrusted images, it could cause the application to disclose sensitive information, crash or, potentially, execute arbitrary code with the Python interpreter's privileges. (CVE-2007-4965, CVE-2008-4864)\n\nMultiple integer underflow and overflow flaws were found in the Python snprintf() wrapper implementation. An attacker could use these flaws to cause a denial of service (memory corruption). (CVE-2008-3144)\n\nMultiple integer overflow flaws were found in various Python modules.\nAn attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-2315, CVE-2008-3143)\n\nAn integer signedness error, leading to a buffer overflow, was found in the Python zlib extension module. If a Python application requested the negative byte count be flushed for a decompression stream, it could cause the application to crash or, potentially, execute arbitrary code with the Python interpreter's privileges.\n(CVE-2008-1721)\n\nA flaw was discovered in the strxfrm() function of the Python locale module. Strings generated by this function were not properly NULL-terminated, which could possibly cause disclosure of data stored in the memory of a Python application using this function.\n(CVE-2007-2052)\n\nRed Hat would like to thank David Remahl of the Apple Product Security team for responsibly reporting the CVE-2008-2315 issue.\n\nAll Python users should upgrade to these updated packages, which contain backported patches to correct these issues.",
            "last seen": "2019-02-21",
            "modified": "2018-11-10",
            "plugin id": "43771",
            "published": "2010-01-06",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=43771",
            "title": "CentOS 5 : python (CESA-2009:1176)"
        },
        {
            "NASL family": "Oracle Linux Local Security Checks",
            "NASL id": "ORACLELINUX_ELSA-2009-1177.NASL",
            "description": "From Red Hat Security Advisory 2009:1177 :\n\nUpdated python packages that fix multiple security issues are now available for Red Hat Enterprise Linux 4.\n\nThis update has been rated as having moderate security impact by the Red Hat Security Response Team.\n\nPython is an interpreted, interactive, object-oriented programming language.\n\nWhen the assert() system call was disabled, an input sanitization flaw was revealed in the Python string object implementation that led to a buffer overflow. The missing check for negative size values meant the Python memory allocator could allocate less memory than expected. This could result in arbitrary code execution with the Python interpreter's privileges. (CVE-2008-1887)\n\nMultiple buffer and integer overflow flaws were found in the Python Unicode string processing and in the Python Unicode and string object implementations. An attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-3142, CVE-2008-5031)\n\nMultiple integer overflow flaws were found in the Python imageop module. If a Python application used the imageop module to process untrusted images, it could cause the application to crash or, potentially, execute arbitrary code with the Python interpreter's privileges. (CVE-2008-1679, CVE-2008-4864)\n\nMultiple integer underflow and overflow flaws were found in the Python snprintf() wrapper implementation. An attacker could use these flaws to cause a denial of service (memory corruption). (CVE-2008-3144)\n\nMultiple integer overflow flaws were found in various Python modules.\nAn attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-2315, CVE-2008-3143)\n\nAn integer signedness error, leading to a buffer overflow, was found in the Python zlib extension module. If a Python application requested the negative byte count be flushed for a decompression stream, it could cause the application to crash or, potentially, execute arbitrary code with the Python interpreter's privileges.\n(CVE-2008-1721)\n\nRed Hat would like to thank David Remahl of the Apple Product Security team for responsibly reporting the CVE-2008-1679 and CVE-2008-2315 issues.\n\nAll Python users should upgrade to these updated packages, which contain backported patches to correct these issues.",
            "last seen": "2019-02-21",
            "modified": "2018-07-18",
            "plugin id": "67897",
            "published": "2013-07-12",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=67897",
            "title": "Oracle Linux 4 : python (ELSA-2009-1177)"
        },
        {
            "NASL family": "Mandriva Local Security Checks",
            "NASL id": "MANDRIVA_MDVSA-2009-003.NASL",
            "description": "Multiple integer overflows in imageop.c in the imageop module in Python 1.5.2 through 2.5.1 allow context-dependent attackers to break out of the Python VM and execute arbitrary code via large integer values in certain arguments to the crop function, leading to a buffer overflow, a different vulnerability than CVE-2007-4965 and CVE-2008-1679. (CVE-2008-4864)\n\nMultiple integer overflows in Python 2.2.3 through 2.5.1, and 2.6, allow context-dependent attackers to have an unknown impact via a large integer value in the tabsize argument to the expandtabs method, as implemented by (1) the string_expandtabs function in Objects/stringobject.c and (2) the unicode_expandtabs function in Objects/unicodeobject.c. NOTE: this vulnerability reportedly exists because of an incomplete fix for CVE-2008-2315. (CVE-2008-5031)\n\nThe updated Python packages have been patched to correct these issues.",
            "last seen": "2019-02-21",
            "modified": "2018-07-19",
            "plugin id": "36693",
            "published": "2009-04-23",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=36693",
            "title": "Mandriva Linux Security Advisory : python (MDVSA-2009:003)"
        },
        {
            "NASL family": "Red Hat Local Security Checks",
            "NASL id": "REDHAT-RHSA-2009-1178.NASL",
            "description": "Updated python packages that fix multiple security issues are now available for Red Hat Enterprise Linux 3.\n\nThis update has been rated as having moderate security impact by the Red Hat Security Response Team.\n\nPython is an interpreted, interactive, object-oriented programming language.\n\nWhen the assert() system call was disabled, an input sanitization flaw was revealed in the Python string object implementation that led to a buffer overflow. The missing check for negative size values meant the Python memory allocator could allocate less memory than expected. This could result in arbitrary code execution with the Python interpreter's privileges. (CVE-2008-1887)\n\nMultiple buffer and integer overflow flaws were found in the Python Unicode string processing and in the Python Unicode and string object implementations. An attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-3142, CVE-2008-5031)\n\nMultiple integer overflow flaws were found in the Python imageop module. If a Python application used the imageop module to process untrusted images, it could cause the application to crash or, potentially, execute arbitrary code with the Python interpreter's privileges. (CVE-2008-1679, CVE-2008-4864)\n\nMultiple integer underflow and overflow flaws were found in the Python snprintf() wrapper implementation. An attacker could use these flaws to cause a denial of service (memory corruption). (CVE-2008-3144)\n\nMultiple integer overflow flaws were found in various Python modules.\nAn attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-2315, CVE-2008-3143)\n\nRed Hat would like to thank David Remahl of the Apple Product Security team for responsibly reporting the CVE-2008-1679 and CVE-2008-2315 issues.\n\nAll Python users should upgrade to these updated packages, which contain backported patches to correct these issues.",
            "last seen": "2019-02-21",
            "modified": "2018-11-27",
            "plugin id": "40402",
            "published": "2009-07-28",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=40402",
            "title": "RHEL 3 : python (RHSA-2009:1178)"
        },
        {
            "NASL family": "CentOS Local Security Checks",
            "NASL id": "CENTOS_RHSA-2009-1178.NASL",
            "description": "Updated python packages that fix multiple security issues are now available for Red Hat Enterprise Linux 3.\n\nThis update has been rated as having moderate security impact by the Red Hat Security Response Team.\n\nPython is an interpreted, interactive, object-oriented programming language.\n\nWhen the assert() system call was disabled, an input sanitization flaw was revealed in the Python string object implementation that led to a buffer overflow. The missing check for negative size values meant the Python memory allocator could allocate less memory than expected. This could result in arbitrary code execution with the Python interpreter's privileges. (CVE-2008-1887)\n\nMultiple buffer and integer overflow flaws were found in the Python Unicode string processing and in the Python Unicode and string object implementations. An attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-3142, CVE-2008-5031)\n\nMultiple integer overflow flaws were found in the Python imageop module. If a Python application used the imageop module to process untrusted images, it could cause the application to crash or, potentially, execute arbitrary code with the Python interpreter's privileges. (CVE-2008-1679, CVE-2008-4864)\n\nMultiple integer underflow and overflow flaws were found in the Python snprintf() wrapper implementation. An attacker could use these flaws to cause a denial of service (memory corruption). (CVE-2008-3144)\n\nMultiple integer overflow flaws were found in various Python modules.\nAn attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-2315, CVE-2008-3143)\n\nRed Hat would like to thank David Remahl of the Apple Product Security team for responsibly reporting the CVE-2008-1679 and CVE-2008-2315 issues.\n\nAll Python users should upgrade to these updated packages, which contain backported patches to correct these issues.",
            "last seen": "2019-02-21",
            "modified": "2018-11-10",
            "plugin id": "40394",
            "published": "2009-07-28",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=40394",
            "title": "CentOS 3 : python (CESA-2009:1178)"
        },
        {
            "NASL family": "Scientific Linux Local Security Checks",
            "NASL id": "SL_20090728_PYTHON_FOR_SL_3_0_X.NASL",
            "description": "When the assert() system call was disabled, an input sanitization flaw was revealed in the Python string object implementation that led to a buffer overflow. The missing check for negative size values meant the Python memory allocator could allocate less memory than expected. This could result in arbitrary code execution with the Python interpreter's privileges. (CVE-2008-1887)\n\nMultiple buffer and integer overflow flaws were found in the Python Unicode string processing and in the Python Unicode and string object implementations. An attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-3142, CVE-2008-5031)\n\nMultiple integer overflow flaws were found in the Python imageop module. If a Python application used the imageop module to process untrusted images, it could cause the application to crash or, potentially, execute arbitrary code with the Python interpreter's privileges. (CVE-2008-1679, CVE-2008-4864)\n\nMultiple integer underflow and overflow flaws were found in the Python snprintf() wrapper implementation. An attacker could use these flaws to cause a denial of service (memory corruption). (CVE-2008-3144)\n\nMultiple integer overflow flaws were found in various Python modules.\nAn attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-2315, CVE-2008-3143)\n\nWould like to thank David Remahl of the Apple Product Security team for responsibly reporting the CVE-2008-1679 and CVE-2008-2315 issues.",
            "last seen": "2019-02-21",
            "modified": "2019-01-02",
            "plugin id": "60624",
            "published": "2012-08-01",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=60624",
            "title": "Scientific Linux Security Update : python for SL 3.0.x on i386/x86_64"
        },
        {
            "NASL family": "Red Hat Local Security Checks",
            "NASL id": "REDHAT-RHSA-2009-1176.NASL",
            "description": "Updated python packages that fix multiple security issues are now available for Red Hat Enterprise Linux 5.\n\nThis update has been rated as having moderate security impact by the Red Hat Security Response Team.\n\nPython is an interpreted, interactive, object-oriented programming language.\n\nWhen the assert() system call was disabled, an input sanitization flaw was revealed in the Python string object implementation that led to a buffer overflow. The missing check for negative size values meant the Python memory allocator could allocate less memory than expected. This could result in arbitrary code execution with the Python interpreter's privileges. (CVE-2008-1887)\n\nMultiple buffer and integer overflow flaws were found in the Python Unicode string processing and in the Python Unicode and string object implementations. An attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-3142, CVE-2008-5031)\n\nMultiple integer overflow flaws were found in the Python imageop module. If a Python application used the imageop module to process untrusted images, it could cause the application to disclose sensitive information, crash or, potentially, execute arbitrary code with the Python interpreter's privileges. (CVE-2007-4965, CVE-2008-4864)\n\nMultiple integer underflow and overflow flaws were found in the Python snprintf() wrapper implementation. An attacker could use these flaws to cause a denial of service (memory corruption). (CVE-2008-3144)\n\nMultiple integer overflow flaws were found in various Python modules.\nAn attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-2315, CVE-2008-3143)\n\nAn integer signedness error, leading to a buffer overflow, was found in the Python zlib extension module. If a Python application requested the negative byte count be flushed for a decompression stream, it could cause the application to crash or, potentially, execute arbitrary code with the Python interpreter's privileges.\n(CVE-2008-1721)\n\nA flaw was discovered in the strxfrm() function of the Python locale module. Strings generated by this function were not properly NULL-terminated, which could possibly cause disclosure of data stored in the memory of a Python application using this function.\n(CVE-2007-2052)\n\nRed Hat would like to thank David Remahl of the Apple Product Security team for responsibly reporting the CVE-2008-2315 issue.\n\nAll Python users should upgrade to these updated packages, which contain backported patches to correct these issues.",
            "last seen": "2019-02-21",
            "modified": "2018-11-27",
            "plugin id": "40400",
            "published": "2009-07-28",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=40400",
            "title": "RHEL 5 : python (RHSA-2009:1176)"
        },
        {
            "NASL family": "Oracle Linux Local Security Checks",
            "NASL id": "ORACLELINUX_ELSA-2009-1176.NASL",
            "description": "From Red Hat Security Advisory 2009:1176 :\n\nUpdated python packages that fix multiple security issues are now available for Red Hat Enterprise Linux 5.\n\nThis update has been rated as having moderate security impact by the Red Hat Security Response Team.\n\nPython is an interpreted, interactive, object-oriented programming language.\n\nWhen the assert() system call was disabled, an input sanitization flaw was revealed in the Python string object implementation that led to a buffer overflow. The missing check for negative size values meant the Python memory allocator could allocate less memory than expected. This could result in arbitrary code execution with the Python interpreter's privileges. (CVE-2008-1887)\n\nMultiple buffer and integer overflow flaws were found in the Python Unicode string processing and in the Python Unicode and string object implementations. An attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-3142, CVE-2008-5031)\n\nMultiple integer overflow flaws were found in the Python imageop module. If a Python application used the imageop module to process untrusted images, it could cause the application to disclose sensitive information, crash or, potentially, execute arbitrary code with the Python interpreter's privileges. (CVE-2007-4965, CVE-2008-4864)\n\nMultiple integer underflow and overflow flaws were found in the Python snprintf() wrapper implementation. An attacker could use these flaws to cause a denial of service (memory corruption). (CVE-2008-3144)\n\nMultiple integer overflow flaws were found in various Python modules.\nAn attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-2315, CVE-2008-3143)\n\nAn integer signedness error, leading to a buffer overflow, was found in the Python zlib extension module. If a Python application requested the negative byte count be flushed for a decompression stream, it could cause the application to crash or, potentially, execute arbitrary code with the Python interpreter's privileges.\n(CVE-2008-1721)\n\nA flaw was discovered in the strxfrm() function of the Python locale module. Strings generated by this function were not properly NULL-terminated, which could possibly cause disclosure of data stored in the memory of a Python application using this function.\n(CVE-2007-2052)\n\nRed Hat would like to thank David Remahl of the Apple Product Security team for responsibly reporting the CVE-2008-2315 issue.\n\nAll Python users should upgrade to these updated packages, which contain backported patches to correct these issues.",
            "last seen": "2019-02-21",
            "modified": "2018-07-18",
            "plugin id": "67896",
            "published": "2013-07-12",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=67896",
            "title": "Oracle Linux 5 : python (ELSA-2009-1176)"
        },
        {
            "NASL family": "Scientific Linux Local Security Checks",
            "NASL id": "SL_20090727_PYTHON_FOR_SL5_X.NASL",
            "description": "When the assert() system call was disabled, an input sanitization flaw was revealed in the Python string object implementation that led to a buffer overflow. The missing check for negative size values meant the Python memory allocator could allocate less memory than expected. This could result in arbitrary code execution with the Python interpreter's privileges. (CVE-2008-1887)\n\nMultiple buffer and integer overflow flaws were found in the Python Unicode string processing and in the Python Unicode and string object implementations. An attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-3142, CVE-2008-5031)\n\nMultiple integer overflow flaws were found in the Python imageop module. If a Python application used the imageop module to process untrusted images, it could cause the application to disclose sensitive information, crash or, potentially, execute arbitrary code with the Python interpreter's privileges. (CVE-2007-4965, CVE-2008-4864)\n\nMultiple integer underflow and overflow flaws were found in the Python snprintf() wrapper implementation. An attacker could use these flaws to cause a denial of service (memory corruption). (CVE-2008-3144)\n\nMultiple integer overflow flaws were found in various Python modules.\nAn attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-2315, CVE-2008-3143)\n\nAn integer signedness error, leading to a buffer overflow, was found in the Python zlib extension module. If a Python application requested the negative byte count be flushed for a decompression stream, it could cause the application to crash or, potentially, execute arbitrary code with the Python interpreter's privileges.\n(CVE-2008-1721)\n\nA flaw was discovered in the strxfrm() function of the Python locale module. Strings generated by this function were not properly NULL-terminated, which could possibly cause disclosure of data stored in the memory of a Python application using this function.\n(CVE-2007-2052)\n\nWould like to thank David Remahl of the Apple Product Security team for responsibly reporting the CVE-2008-2315 issue.",
            "last seen": "2019-02-21",
            "modified": "2019-01-02",
            "plugin id": "60622",
            "published": "2012-08-01",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=60622",
            "title": "Scientific Linux Security Update : python for SL5.x i386/x86_64"
        },
        {
            "NASL family": "Red Hat Local Security Checks",
            "NASL id": "REDHAT-RHSA-2009-1177.NASL",
            "description": "Updated python packages that fix multiple security issues are now available for Red Hat Enterprise Linux 4.\n\nThis update has been rated as having moderate security impact by the Red Hat Security Response Team.\n\nPython is an interpreted, interactive, object-oriented programming language.\n\nWhen the assert() system call was disabled, an input sanitization flaw was revealed in the Python string object implementation that led to a buffer overflow. The missing check for negative size values meant the Python memory allocator could allocate less memory than expected. This could result in arbitrary code execution with the Python interpreter's privileges. (CVE-2008-1887)\n\nMultiple buffer and integer overflow flaws were found in the Python Unicode string processing and in the Python Unicode and string object implementations. An attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-3142, CVE-2008-5031)\n\nMultiple integer overflow flaws were found in the Python imageop module. If a Python application used the imageop module to process untrusted images, it could cause the application to crash or, potentially, execute arbitrary code with the Python interpreter's privileges. (CVE-2008-1679, CVE-2008-4864)\n\nMultiple integer underflow and overflow flaws were found in the Python snprintf() wrapper implementation. An attacker could use these flaws to cause a denial of service (memory corruption). (CVE-2008-3144)\n\nMultiple integer overflow flaws were found in various Python modules.\nAn attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-2315, CVE-2008-3143)\n\nAn integer signedness error, leading to a buffer overflow, was found in the Python zlib extension module. If a Python application requested the negative byte count be flushed for a decompression stream, it could cause the application to crash or, potentially, execute arbitrary code with the Python interpreter's privileges.\n(CVE-2008-1721)\n\nRed Hat would like to thank David Remahl of the Apple Product Security team for responsibly reporting the CVE-2008-1679 and CVE-2008-2315 issues.\n\nAll Python users should upgrade to these updated packages, which contain backported patches to correct these issues.",
            "last seen": "2019-02-21",
            "modified": "2018-11-27",
            "plugin id": "40401",
            "published": "2009-07-28",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=40401",
            "title": "RHEL 4 : python (RHSA-2009:1177)"
        },
        {
            "NASL family": "Oracle Linux Local Security Checks",
            "NASL id": "ORACLELINUX_ELSA-2009-1178.NASL",
            "description": "From Red Hat Security Advisory 2009:1178 :\n\nUpdated python packages that fix multiple security issues are now available for Red Hat Enterprise Linux 3.\n\nThis update has been rated as having moderate security impact by the Red Hat Security Response Team.\n\nPython is an interpreted, interactive, object-oriented programming language.\n\nWhen the assert() system call was disabled, an input sanitization flaw was revealed in the Python string object implementation that led to a buffer overflow. The missing check for negative size values meant the Python memory allocator could allocate less memory than expected. This could result in arbitrary code execution with the Python interpreter's privileges. (CVE-2008-1887)\n\nMultiple buffer and integer overflow flaws were found in the Python Unicode string processing and in the Python Unicode and string object implementations. An attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-3142, CVE-2008-5031)\n\nMultiple integer overflow flaws were found in the Python imageop module. If a Python application used the imageop module to process untrusted images, it could cause the application to crash or, potentially, execute arbitrary code with the Python interpreter's privileges. (CVE-2008-1679, CVE-2008-4864)\n\nMultiple integer underflow and overflow flaws were found in the Python snprintf() wrapper implementation. An attacker could use these flaws to cause a denial of service (memory corruption). (CVE-2008-3144)\n\nMultiple integer overflow flaws were found in various Python modules.\nAn attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-2315, CVE-2008-3143)\n\nRed Hat would like to thank David Remahl of the Apple Product Security team for responsibly reporting the CVE-2008-1679 and CVE-2008-2315 issues.\n\nAll Python users should upgrade to these updated packages, which contain backported patches to correct these issues.",
            "last seen": "2019-02-21",
            "modified": "2018-07-18",
            "plugin id": "67898",
            "published": "2013-07-12",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=67898",
            "title": "Oracle Linux 3 : python (ELSA-2009-1178)"
        },
        {
            "NASL family": "Scientific Linux Local Security Checks",
            "NASL id": "SL_20090728_PYTHON_FOR_SL_4_X.NASL",
            "description": "When the assert() system call was disabled, an input sanitization flaw was revealed in the Python string object implementation that led to a buffer overflow. The missing check for negative size values meant the Python memory allocator could allocate less memory than expected. This could result in arbitrary code execution with the Python interpreter's privileges. (CVE-2008-1887)\n\nMultiple buffer and integer overflow flaws were found in the Python Unicode string processing and in the Python Unicode and string object implementations. An attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-3142, CVE-2008-5031)\n\nMultiple integer overflow flaws were found in the Python imageop module. If a Python application used the imageop module to process untrusted images, it could cause the application to crash or, potentially, execute arbitrary code with the Python interpreter's privileges. (CVE-2008-1679, CVE-2008-4864)\n\nMultiple integer underflow and overflow flaws were found in the Python snprintf() wrapper implementation. An attacker could use these flaws to cause a denial of service (memory corruption). (CVE-2008-3144)\n\nMultiple integer overflow flaws were found in various Python modules.\nAn attacker could use these flaws to cause a denial of service (Python application crash). (CVE-2008-2315, CVE-2008-3143)\n\nAn integer signedness error, leading to a buffer overflow, was found in the Python zlib extension module. If a Python application requested the negative byte count be flushed for a decompression stream, it could cause the application to crash or, potentially, execute arbitrary code with the Python interpreter's privileges.\n(CVE-2008-1721)\n\nWould like to thank David Remahl of the Apple Product Security team for responsibly reporting the CVE-2008-1679 and CVE-2008-2315 issues.",
            "last seen": "2019-02-21",
            "modified": "2019-01-02",
            "plugin id": "60625",
            "published": "2012-08-01",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=60625",
            "title": "Scientific Linux Security Update : python for SL 4.x on i386/x86_64"
        }
    ],
    "oval": [
        {
            "accepted": "2013-04-29T04:07:54.291-04:00",
            "class": "vulnerability",
            "contributors": [
                {
                    "name": "Aharon Chernin",
                    "organization": "SCAP.com, LLC"
                },
                {
                    "name": "Dragos Prisaca",
                    "organization": "G2, Inc."
                }
            ],
            "definition_extensions": [
                {
                    "comment": "The operating system installed on the system is Red Hat Enterprise Linux 3",
                    "oval": "oval:org.mitre.oval:def:11782"
                },
                {
                    "comment": "CentOS Linux 3.x",
                    "oval": "oval:org.mitre.oval:def:16651"
                },
                {
                    "comment": "The operating system installed on the system is Red Hat Enterprise Linux 4",
                    "oval": "oval:org.mitre.oval:def:11831"
                },
                {
                    "comment": "CentOS Linux 4.x",
                    "oval": "oval:org.mitre.oval:def:16636"
                },
                {
                    "comment": "Oracle Linux 4.x",
                    "oval": "oval:org.mitre.oval:def:15990"
                },
                {
                    "comment": "The operating system installed on the system is Red Hat Enterprise Linux 5",
                    "oval": "oval:org.mitre.oval:def:11414"
                },
                {
                    "comment": "The operating system installed on the system is CentOS Linux 5.x",
                    "oval": "oval:org.mitre.oval:def:15802"
                },
                {
                    "comment": "Oracle Linux 5.x",
                    "oval": "oval:org.mitre.oval:def:15459"
                }
            ],
            "description": "Multiple integer overflows in imageop.c in the imageop module in Python 1.5.2 through 2.5.1 allow context-dependent attackers to break out of the Python VM and execute arbitrary code via large integer values in certain arguments to the crop function, leading to a buffer overflow, a different vulnerability than CVE-2007-4965 and CVE-2008-1679.",
            "family": "unix",
            "id": "oval:org.mitre.oval:def:10702",
            "status": "accepted",
            "submitted": "2010-07-09T03:56:16-04:00",
            "title": "Multiple integer overflows in imageop.c in the imageop module in Python 1.5.2 through 2.5.1 allow context-dependent attackers to break out of the Python VM and execute arbitrary code via large integer values in certain arguments to the crop function, leading to a buffer overflow, a different vulnerability than CVE-2007-4965 and CVE-2008-1679.",
            "version": "24"
        },
        {
            "accepted": "2014-01-20T04:01:38.702-05:00",
            "class": "vulnerability",
            "contributors": [
                {
                    "name": "Pai Peng",
                    "organization": "Hewlett-Packard"
                },
                {
                    "name": "Chris Coffin",
                    "organization": "The MITRE Corporation"
                }
            ],
            "definition_extensions": [
                {
                    "comment": "VMWare ESX Server 3.0.3 is installed",
                    "oval": "oval:org.mitre.oval:def:6026"
                },
                {
                    "comment": "VMware ESX Server 3.5.0 is installed",
                    "oval": "oval:org.mitre.oval:def:5887"
                },
                {
                    "comment": "VMware ESX Server 4.0 is installed",
                    "oval": "oval:org.mitre.oval:def:6293"
                }
            ],
            "description": "Multiple integer overflows in imageop.c in the imageop module in Python 1.5.2 through 2.5.1 allow context-dependent attackers to break out of the Python VM and execute arbitrary code via large integer values in certain arguments to the crop function, leading to a buffer overflow, a different vulnerability than CVE-2007-4965 and CVE-2008-1679.",
            "family": "unix",
            "id": "oval:org.mitre.oval:def:8354",
            "status": "accepted",
            "submitted": "2010-03-19T16:57:59.000-04:00",
            "title": "VMware python multiple integer overflows vulnerability in the imageop module",
            "version": "7"
        }
    ],
    "ranking": [
        [
            {
                "circl": 3
            }
        ]
    ],
    "redhat": {
        "rpms": [
            "python-0:2.4.3-24.el5_3.6",
            "python-devel-0:2.4.3-24.el5_3.6",
            "python-tools-0:2.4.3-24.el5_3.6",
            "tkinter-0:2.4.3-24.el5_3.6",
            "python-0:2.3.4-14.7.el4_8.2",
            "python-devel-0:2.3.4-14.7.el4_8.2",
            "python-docs-0:2.3.4-14.7.el4_8.2",
            "python-tools-0:2.3.4-14.7.el4_8.2",
            "tkinter-0:2.3.4-14.7.el4_8.2",
            "python-0:2.2.3-6.11",
            "python-devel-0:2.2.3-6.11",
            "python-tools-0:2.2.3-6.11",
            "tkinter-0:2.2.3-6.11"
        ]
    },
    "references": [
        "http://lists.apple.com/archives/security-announce/2009/Feb/msg00000.html",
        "http://scary.beasts.org/security/CESA-2008-008.html",
        "http://support.apple.com/kb/HT3438",
        "http://svn.python.org/view/python/trunk/Modules/imageop.c?rev=66689&view=diff&r1=66689&r2=66688&p1=python/trunk/Modules/imageop.c&p2=/python/trunk/Modules/imageop.c",
        "http://svn.python.org/view?rev=66689&view=rev",
        "http://www.openwall.com/lists/oss-security/2008/10/27/2",
        "http://www.openwall.com/lists/oss-security/2008/10/29/3",
        "http://www.securityfocus.com/archive/1/507985/100/0/threaded",
        "http://www.securityfocus.com/bid/31932",
        "http://www.securityfocus.com/bid/31976",
        "http://www.vmware.com/security/advisories/VMSA-2009-0016.html",
        "http://www.vupen.com/english/advisories/2009/3316",
        "https://exchange.xforce.ibmcloud.com/vulnerabilities/46606"
    ],
    "refmap": {
        "apple": [
            "APPLE-SA-2009-02-12"
        ],
        "bid": [
            "31932",
            "31976"
        ],
        "bugtraq": [
            "20091120 VMSA-2009-0016 VMware vCenter and ESX update release and vMA patch release address multiple security issue in third party components"
        ],
        "confirm": [
            "http://support.apple.com/kb/HT3438",
            "http://svn.python.org/view/python/trunk/Modules/imageop.c?rev=66689&view=diff&r1=66689&r2=66688&p1=python/trunk/Modules/imageop.c&p2=/python/trunk/Modules/imageop.c",
            "http://svn.python.org/view?rev=66689&view=rev",
            "http://www.vmware.com/security/advisories/VMSA-2009-0016.html"
        ],
        "misc": [
            "http://scary.beasts.org/security/CESA-2008-008.html"
        ],
        "mlist": [
            "[oss-security] 20081027 CVE request -- Python imageop#3",
            "[oss-security] 20081029 CVE Request - Python imageop"
        ],
        "secunia": [
            "33937",
            "37471"
        ],
        "vupen": [
            "ADV-2009-3316"
        ],
        "xf": [
            "python-image-module-bo(46606)"
        ]
    },
    "summary": "Multiple integer overflows in imageop.c in the imageop module in Python 1.5.2 through 2.5.1 allow context-dependent attackers to break out of the Python VM and execute arbitrary code via large integer values in certain arguments to the crop function, leading to a buffer overflow, a different vulnerability than CVE-2007-4965 and CVE-2008-1679.",
    "vulnerable_configuration": [
        {
            "id": "cpe:2.3:a:python_software_foundation:python:1.5.2",
            "title": "cpe:2.3:a:python_software_foundation:python:1.5.2"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:1.6",
            "title": "cpe:2.3:a:python_software_foundation:python:1.6"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:1.6.1",
            "title": "cpe:2.3:a:python_software_foundation:python:1.6.1"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.0",
            "title": "cpe:2.3:a:python_software_foundation:python:2.0"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.0.1",
            "title": "cpe:2.3:a:python_software_foundation:python:2.0.1"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.1",
            "title": "cpe:2.3:a:python_software_foundation:python:2.1"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.1.1",
            "title": "cpe:2.3:a:python_software_foundation:python:2.1.1"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.1.2",
            "title": "cpe:2.3:a:python_software_foundation:python:2.1.2"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.1.3",
            "title": "cpe:2.3:a:python_software_foundation:python:2.1.3"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.2",
            "title": "cpe:2.3:a:python_software_foundation:python:2.2"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.2.1",
            "title": "cpe:2.3:a:python_software_foundation:python:2.2.1"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.2.2",
            "title": "cpe:2.3:a:python_software_foundation:python:2.2.2"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.2.3",
            "title": "cpe:2.3:a:python_software_foundation:python:2.2.3"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.3",
            "title": "cpe:2.3:a:python_software_foundation:python:2.3"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.3.1",
            "title": "cpe:2.3:a:python_software_foundation:python:2.3.1"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.3.2",
            "title": "cpe:2.3:a:python_software_foundation:python:2.3.2"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.3.3",
            "title": "cpe:2.3:a:python_software_foundation:python:2.3.3"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.3.4",
            "title": "cpe:2.3:a:python_software_foundation:python:2.3.4"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.3.5",
            "title": "cpe:2.3:a:python_software_foundation:python:2.3.5"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.3.6",
            "title": "cpe:2.3:a:python_software_foundation:python:2.3.6"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.4",
            "title": "cpe:2.3:a:python_software_foundation:python:2.4"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.4.1",
            "title": "cpe:2.3:a:python_software_foundation:python:2.4.1"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.4.2",
            "title": "cpe:2.3:a:python_software_foundation:python:2.4.2"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.4.3",
            "title": "cpe:2.3:a:python_software_foundation:python:2.4.3"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.4.4",
            "title": "cpe:2.3:a:python_software_foundation:python:2.4.4"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.5",
            "title": "cpe:2.3:a:python_software_foundation:python:2.5"
        },
        {
            "id": "cpe:2.3:a:python_software_foundation:python:2.5.1",
            "title": "cpe:2.3:a:python_software_foundation:python:2.5.1"
        }
    ],
    "vulnerable_configuration_cpe_2_2": [
        "cpe:/a:python_software_foundation:python:1.5.2",
        "cpe:/a:python_software_foundation:python:1.6",
        "cpe:/a:python_software_foundation:python:1.6.1",
        "cpe:/a:python_software_foundation:python:2.0",
        "cpe:/a:python_software_foundation:python:2.0.1",
        "cpe:/a:python_software_foundation:python:2.1",
        "cpe:/a:python_software_foundation:python:2.1.1",
        "cpe:/a:python_software_foundation:python:2.1.2",
        "cpe:/a:python_software_foundation:python:2.1.3",
        "cpe:/a:python_software_foundation:python:2.2",
        "cpe:/a:python_software_foundation:python:2.2.1",
        "cpe:/a:python_software_foundation:python:2.2.2",
        "cpe:/a:python_software_foundation:python:2.2.3",
        "cpe:/a:python_software_foundation:python:2.3",
        "cpe:/a:python_software_foundation:python:2.3.1",
        "cpe:/a:python_software_foundation:python:2.3.2",
        "cpe:/a:python_software_foundation:python:2.3.3",
        "cpe:/a:python_software_foundation:python:2.3.4",
        "cpe:/a:python_software_foundation:python:2.3.5",
        "cpe:/a:python_software_foundation:python:2.3.6",
        "cpe:/a:python_software_foundation:python:2.4",
        "cpe:/a:python_software_foundation:python:2.4.1",
        "cpe:/a:python_software_foundation:python:2.4.2",
        "cpe:/a:python_software_foundation:python:2.4.3",
        "cpe:/a:python_software_foundation:python:2.4.4",
        "cpe:/a:python_software_foundation:python:2.5",
        "cpe:/a:python_software_foundation:python:2.5.1"
    ]
}